src/next_array.rs

```diff
@@ -0,0 +1,64 @@
+use core::mem::MaybeUninit;
```

Comment:
This lets the later invocation of `core::mem::replace` be a little more brief:
```suggestion
use core::mem::{self, MaybeUninit};
```